arrayfire_wrapper/_backend.py

```diff
@@ -108,26 +159,47 @@ def _find_default_path(*args: str) -> Path:
 
 
 class BackendType(enum.Enum):  # TODO change name - avoid using _backend_type - e.g. type
-    unified = 0  # NOTE It is set as Default value on Arrayfire backend
-    cpu = 1
     cuda = 2
     opencl = 4
     oneapi = 8
+    cpu = 1
+    unified = 0  # NOTE It is set as Default value on Arrayfire backend
 
     def __iter__(self) -> Iterator:
         # NOTE cpu comes last because we want to keep this order priorty during backend initialization
-        return iter((self.unified, self.cuda, self.oneapi, self.opencl, self.cpu))
+        return iter((self.cuda, self.opencl, self.oneapi, self.cpu, self.unified))
 
 
 class Backend:
     _backend_type: BackendType
-    _clib: ctypes.CDLL
+    _clibs: dict[BackendType, ctypes.CDLL]
 
     def __init__(self) -> None:
         self._backend_path_config = _get_backend_path_config()
 
-        self._load_forge_lib()
+        self._backend_type = None
+        self._clibs = {}
         self._load_backend_libs()
+        self._load_forge_lib()
+
+    def set_backend(self, backend_type : BackendType) -> None:
+        # if unified is available, do dynamic module loading through libaf
+        if self._backend_type == BackendType.unified:
+            from arrayfire_wrapper.lib.unified_api_functions import set_backend as unified_set_backend
+            try:
+                unified_set_backend(backend_type)
+            except RuntimeError:
+                if VERBOSE_LOADS:
+                    print(f"Unable to change backend using unified loader")
+                raise RuntimeError
```

Comment:
Why do you catch RuntimeError to raise it? Why? 😱

Reply:
If a unified version of `set_backend` fails, it will be raised by the safe call and the corresponding error will be thrown. No need to add try/except just to surpass an error from a safe call.